### DISCUSSION TITLE
Templating common elements to be used on multiple pages

### DISCUSSION THREAD

**Discussion by thetreythomas** [zauberzeug/nicegui#985] [Ideas / Feature Requests]
I'm coming from Django and other web frameworks that use Jinja. I'm used to creating a header and footer page that can then be used with the `{% extends %}` tag. This keeps from having to duplicate things in all the other template files.

Looking through the NiceGUI documentation, I don't see a clear way to go about this.

If I am using multiple pages in my NiceGUI application, how would one go about getting the header and footer to show up on every page without duplicating code for the `ui.header` and `ui.footer`?

Also, the example for the [Page Layout](https://nicegui.io/documentation#page_layout), why is `async` used for this?  I don't see the `async` used in most of the other examples, wondering if it's required for these specific UI elements.

**Comment by rodja** [comment]
Thanks for reaching out @thetreythomas. Have you seen our [modularization example](https://github.com/zauberzeug/nicegui/tree/main/examples/modularization)? It demonstrates various ways for code reuse and splitting into separate files.

I've removed the async statement in https://github.com/zauberzeug/nicegui/commit/76edb29194009af8cbfa5f7174550d4c39919791. Thanks for spotting and reporting this incositency.
